### PR TITLE
Account

### PR DESCRIPTION
**Pull request by Nandhini-khoros**
Account details

### CODE REVIEW

File: .idea/uiDesigner.xml
Comment (grewaviv) [inline .idea/uiDesigner.xml]: What is the purpose of this file?

File: Banking/src/main/java/com/example/Banking/controller/AccountController.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/controller/AccountController.java]: "*" import is not recommended. Import only classes which is being used

**Review by grewaviv — CHANGES_REQUESTED** [review]
A Dao class should be responsible for database interactions. Controller should call service which in turn should call Dao for db interactions

File: Banking/src/main/java/com/example/Banking/exception/ResourceNotFoundException.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/exception/ResourceNotFoundException.java]: This is nice! A custom exception. Before curly braces there should be a space "Exception {"

File: Banking/src/main/java/com/example/Banking/model/Account.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/model/Account.java]: The name accountdetails2 doesn't sound good to me. It can be account_details, whenever theres a word break add underscore for table names.

File: Banking/src/main/java/com/example/Banking/model/Account.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/model/Account.java]: I would recommend annotate fields with column names and avoid annotating getters. And instead of adding getters, setters and toString by yourself use Lombok. Lombok has annotations such as @Data which ensures getters and setters without creating it manually. It makes the code clean and more readable.

**Comment by grewaviv** [comment]
While creating the PR please explain the purpose of the PR in the description and if possible add sample API calls and their outputs. If its a for UI adding screenshots. Also adding PR comments wherever necessary for the reviewer to easily understand.

File: .idea/uiDesigner.xml
Comment (Nandhini-khoros) [reply .idea/uiDesigner.xml]: It was there in my branch when I created it. So I thought it's an inbuilt file.

File: Banking/src/main/java/com/example/Banking/controller/AccountController.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/controller/AccountController.java]: Okay

File: Banking/src/main/java/com/example/Banking/exception/ResourceNotFoundException.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/exception/ResourceNotFoundException.java]: Okay

File: Banking/src/main/java/com/example/Banking/model/Account.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/model/Account.java]: Okay

File: Banking/src/main/java/com/example/Banking/model/Account.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/model/Account.java]: Okay

**Comment by Nandhini-khoros** [comment]
> While creating the PR please explain the purpose of the PR in the description and if possible add sample API calls and their outputs. If its a for UI adding screenshots. Also adding PR comments wherever necessary for the reviewer to easily understand.

Okay

**Comment by Nandhini-khoros** [comment]
> A Dao class should be responsible for database interactions. Controller should call service which in turn should call Dao for db interactions

Okay

File: .idea/uiDesigner.xml
Comment (grewaviv) [reply .idea/uiDesigner.xml]: No such file should be committed to git as these are intellij specific files. Please add .gitignore to ignore any files in .idea folder. If its created in git please delete it

Comment (Nandhini-khoros) [reply .idea/uiDesigner.xml]: Okay

**Comment by gokul-khoros** [comment]
> > While creating the PR please explain the purpose of the PR in the description and if possible add sample API calls and their outputs. If its a for UI adding screenshots. Also adding PR comments wherever necessary for the reviewer to easily understand.
> 
> Okay

File: Banking/src/main/java/com/example/Banking/controller/AccountController.java
Comment (grewaviv) [reply Banking/src/main/java/com/example/Banking/controller/AccountController.java]: This still needs to be fixed

File: Banking/build.gradle
Comment (grewaviv) [inline Banking/build.gradle]: Follow proper indentations. Best practice is not to hardcode the versions and to have the versions as variables in a separate file, lets say depVersions.gradle which only contains variables representing the version numbers of the dependencies.

This will keep the versions at one place which can be reused in multiple modules and also easy to change.

File: Banking/src/main/java/com/example/Banking/controller/AccountController.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/controller/AccountController.java]: I prefer annotations to be just above class/method/variables without any space, its easier to read otherwise anyone can miss the annotation

File: Banking/src/main/java/com/example/Banking/service/Service.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/Service.java]: Its name should be `AccountService`

File: Banking/src/main/java/com/example/Banking/exception/ErrorHandler.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/exception/ErrorHandler.java]: Shouldn't be space between class and its corresponding annotation

File: Banking/src/main/java/com/example/Banking/exception/IdNotFound.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/exception/IdNotFound.java]: It should be `IdNotFoundException`

File: Banking/src/main/java/com/example/Banking/exception/IdNotFound.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/exception/IdNotFound.java]: You can use `@AllArgsConstructor` at the top of the class to avoid creating constructor by yourself

File: Banking/src/main/java/com/example/Banking/model/Account.java
Comment (grewaviv) [reply Banking/src/main/java/com/example/Banking/model/Account.java]: You did not mention which column represents which field. Also follow camel casing while naming a field in java. Field names should start with lowercases. For example `accountNumber`

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: You don't have to add a comment if the method name or code is self explanatory or does not contain business logic. `AccountDaoImpl.getAll` is pretty obvious returning all accounts

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: Exception should also contain the id which is causing the exception. Also do not hardcode the error strings. Create constants for error strings which can be reusable. For example delete and update can have same error strings.

File: Banking/src/main/java/com/example/Banking/service/Service.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/Service.java]: Follow camel case `createAccount`

File: Banking/src/main/java/com/example/Banking/controller/AccountController.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/controller/AccountController.java]: Sure I will correct it.

File: Banking/src/main/java/com/example/Banking/controller/AccountController.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/controller/AccountController.java]: I removed the space and pushed the code

File: Banking/src/main/java/com/example/Banking/service/Service.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/service/Service.java]: I have changed

File: Banking/src/main/java/com/example/Banking/exception/ErrorHandler.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/exception/ErrorHandler.java]: I have removed the space

File: Banking/src/main/java/com/example/Banking/exception/IdNotFound.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/exception/IdNotFound.java]: I have changed it

File: Banking/src/main/java/com/example/Banking/exception/IdNotFound.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/exception/IdNotFound.java]: I have used @AllArgsConstructor and pushed the code

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: Okay

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: I have changed and pushed the code

File: Banking/src/main/java/com/example/Banking/service/Service.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/service/Service.java]: Sure. I will rectify it.

File: Banking/build.gradle
Comment (Nandhini-khoros) [reply Banking/build.gradle]: Sure. I will rectify it.

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: A constant should be declared as static and final with upper cases. There should be space before and after '=' sign for better readability. The variable should be named for the purpose its created. msg doesn't match with the purpose its serving

Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: Sure. I will change it

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: `ERROR_MESSAGE + " "+ acNumber`

File: Banking/src/main/java/com/example/Banking/controller/AccountController.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/controller/AccountController.java]: account has accountNumber. I don't think we need to pass accountNumber separately

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: There are some things which shouldn't be modified by update operation. Like accountNumber which is the primary key

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: I have changed and pushed it

**Comment by grewaviv** [comment]
This PR is almost there. Just incorporate the recent review comments.

File: Banking/src/main/java/com/example/Banking/controller/AccountController.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/controller/AccountController.java]: Only If I pass the account number that particular account details will be fetched for changes to be made

Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/controller/AccountController.java]: I have reomoved the account number and performed the update operation. It is working.

Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/controller/AccountController.java]: removed*

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: accounts should be renamed to something meaningful, like accountFromDB

File: Banking/src/main/java/com/example/Banking/service/AccountService.java
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/service/AccountService.java]: Now since we are not updating on Id the method should be renamed to updateAccount

**Review by grewaviv — APPROVED** [review]
I have some comments on renaming the stuff. You can merge this code after renaming it.

**Comment by grewaviv** [comment]
Make sure to resolve the merge conflicts before merging

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: I have renamed and pushed it.

File: Banking/src/main/java/com/example/Banking/service/AccountService.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/service/AccountService.java]: I have renamed to updateAccount and pushed it

File: Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java
Comment (Nandhini-khoros) [reply Banking/src/main/java/com/example/Banking/service/AccountDaoImpl.java]: > I have some comments on renaming the stuff. You can merge this code after renaming it.
I have renamed the stuff. I will merge.

**Pull request closed without merging**
